.github/workflows/docker.yml

```diff
@@ -49,3 +53,28 @@ jobs:
         tags: ${{ env.REGISTRY }}/nordsecurity/${{ inputs.name }}${{ inputs.rust_version }}:${{ inputs.version }}
         build-args: |
           REVISION=${{ github.sha }}
+
+  build-docker-on-branch:
+    if: github.event_name == 'push'
+    strategy:
+      matrix:
+        name:
+          - build-android-rust
+          - build-linux-rust
+          - build-windows-rust
+          - package-aar
+          - package-aar-jdk-17
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+    steps:
+    - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+    - name: Build and push Docker image
+      uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09 # v5.0.0
+      with:
+        context: .
+        file: builders/${{ matrix.name }}/Dockerfile
+        push: false
```

Comment:
don't we want to push so that it's possible to test those images?

Reply:
I was thinking about it and left if for a debate :) The problem I had was that GH container registry does not support nested paths. E.g. it would be great if we could have them stored in some path like 'nordsecurity/rust_build_utils/branches/<branchame><matrix.name>:latest' but we can't so I'm just worried we would need to pollute the registry with 'nordsecurity/rust_build_utils/<branchname><matrix.name>:latest. I just now thought that maybe we could maybe also use one in branch image name and automatically update tags. I would need to check how easy that is

Reply:
I think that the fact that the image builds is not enough and before reviewing/merging each PR should have a image tested in for example libtelio. To do that you need to have it pushed somewhere and so it's better to automate that. Could we do `nordsecurity/rust_build_utils/<matrix.name>:<branch_name>_latest`?

Reply:
Nice idea. For some reason it didn't occur to me to use branch name in tag part :D